src/kafka-net/ProtocolGateway.cs

```diff
@@ -0,0 +1,139 @@
+﻿using KafkaNet.Model;
+using KafkaNet.Protocol;
+using System;
+using System.Linq;
+using System.Net.Sockets;
+using System.Runtime.ExceptionServices;
+using System.Threading.Tasks;
+
+namespace KafkaNet
+{
+    public class ProtocolGateway : IDisposable
+    {
+        private readonly IBrokerRouter _brokerRouter;
+        //Add Loger
```

Comment:
eran, please go over the code and make sure that you always have a space line before comments